Composer/packages/client/src/pages/publish/Publish.tsx

```diff
@@ -288,8 +282,8 @@ const Publish: React.FC<RouteComponentProps<{ projectId: string; targetName?: st
       if (target && isPublishingToAzure(target)) {
         const { tenantId } = JSON.parse(target.configuration);
 
-        if (userShouldProvideTokens()) {
-          token = getTokenFromCache('accessToken');
+        if (userShouldProvideTokens() && currentUser) {
```

Comment:
Non-blocking issue: I'm hoping that we can remove the call differences so that userShouldProvideTokens isn't called anywhere.  The require login call can inspect if we are in electron or not and bring up the right login prompt or token dialog.

Reply:
good point

Reply:
there are not a ton of places this is called, and many of them are probably not needed anymore.  we should do this separately imo.